main_test.go

```diff
@@ -0,0 +1,287 @@
+	}
+	assert.Equalf(t, handlers.REDIRECT, requestLoginResponse.RedirectTo, "Expected %s, got %s", handlers.REDIRECT, requestLoginResponse.RedirectTo)
+}
+
+func TestHandleUpdateFlow(t *testing.T) {
+	//init clients
+	t.Cleanup(testServers.CreateTestServers())
+
+	//create request
+	body := kratos_client.NewUpdateLoginFlowWithOidcMethod(UPDATE_LOGIN_FLOW_METHOD, UPDATE_LOGIN_FLOW_PROVIDER)
+	bodyJson, err := json.Marshal(*body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	bodyReader := bytes.NewReader(bodyJson)
+	req := httptest.NewRequest(http.MethodPost, HANDLE_UPDATE_LOGIN_FLOW_URL, bodyReader)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-Session-Token", "test-x-session-token")
+	cookie := &http.Cookie{
+		Name:   COOKIE_NAME,
+		Value:  COOKIE_VALUE,
+		MaxAge: 300,
+	}
+	req.AddCookie(cookie)
+
+	//create response
+	w := httptest.NewRecorder()
+	//start function
+	handleUpdateFlow(w, req)
+	//check results
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	loginUpdateResponse := kratos_client.NewSuccessfulNativeLoginWithDefaults()
+	if err := json.Unmarshal(data, loginUpdateResponse); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	assert.Equalf(t, handlers.SESSION_ID, loginUpdateResponse.Session.Id, "Expected %s, got %s", handlers.SESSION_ID, loginUpdateResponse.Session.Id)
+}
+
+func TestHandleKratosError(t *testing.T) {
+	t.Cleanup(testServers.CreateTestServers())
+
+	req := httptest.NewRequest(http.MethodGet, HANDLE_ERROR_URL, nil)
+	w := httptest.NewRecorder()
+	handleKratosError(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	te := new(handlers.TestErrorReport)
+	if err := json.Unmarshal(data, te); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	assert.Equalf(t, handlers.ERROR_MESSAGE, te.Error.Message, "Expected %s, got %s", handlers.ERROR_MESSAGE, te.Error.Message)
+
+}
+
+func TestHandleConsent(t *testing.T) {
+	t.Cleanup(testServers.CreateTestServers())
+
+	req := httptest.NewRequest(http.MethodGet, HANDLE_CONSENT_URL, nil)
+	w := httptest.NewRecorder()
+	handleConsent(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+	responseRedirect := hydra_client.NewOAuth2RedirectToWithDefaults()
+	if err := json.Unmarshal(data, responseRedirect); err != nil {
+		t.Errorf("expected error to be nil got %v", err)
+	}
+
+	if responseRedirect.RedirectTo != "test.test" {
+		t.Errorf("expected test.test, got %v", string(data))
+	}
+	assert.Equalf(t, "test.test", responseRedirect.RedirectTo, "Expected %s, got %s.", "test.test", responseRedirect.RedirectTo)
```

Comment:
it is. I kinda had to flip flop between refactoring the functions because of cross dependencies in the modules (good call for simplifying module layout btw.) This is a mistake. I fixed it.